### ISSUE TITLE
Dynamic image carousel not working with Media Carousel Widget and custom field (ACF and JetEngine)

### ISSUE THREAD

**Issue by ThomasCommpagnie** [elementor/elementor#23224]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Hi,

Following issue #23191, closed after not reproducing correctly the issue, I would like to open a new ticket. On the following video : https://github.com/elementor/elementor/assets/113993460/df52b543-fe0d-4422-bb1a-9ffe793b9bb9, you reproducted the situation but forgot to set the skin on "Diaporama" instead of "Carousel".

I'm experiencing an incompatibility between Elementor Media Carousel Widget and Crocoblock JetEngine dynamic field.

I'm using 4 custom field in order to upload images and query them in my Media Carousel Widget.

It is working well on the top image in "'Diaporama" skin but the widget doesn't get the correct images on the diaporama under the top image.

![Capture d’écran 2023-07-31 104800](https://github.com/elementor/elementor/assets/55954595/fd0e47a8-3fea-4c10-afc0-572287527db3)

### Steps to reproduce

Create a media custom field in JetEngine or ACF

Upload your image on the media custom field

Link the custom field to the elementor widget ; each tab of the widget must have a separate custom field created. For example, if you have 3 media custom field, you must have 3 tabs on your Media Carousel widget.
The carousel must be set on "Diaporama" skin.

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: MariaDB Server v10.6.14
	PHP Version: 7.4.33
	PHP Memory Limit: 4096M
	PHP Max Input Vars: 2000000
	PHP Max Post Size: 512M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.2.2
	Site URL: https://v3.le-rio.fr
	Home URL: https://v3.le-rio.fr
	WP Multisite: No
	Max Upload Size: 512 Mo
	Memory limit: 40M
	Max Memory limit: 4096M
	Permalink Structure: /%year%/%monthnum%/%day%/%postname%/
	Language: fr-FR
	Timezone: Europe/Paris
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor Child
	Version: 2.0.0
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.8.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: fr_FR
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36

== Active Plugins ==
	Disable Gutenberg
		Version: 3.0
		Author: Jeff Starr

	DPE
		Version: 3.1
		Author: EPN

	DynamicConditions
		Version: 1.6.2
		Author: RTO GmbH

	Dynamic Visibility for Elementor
		Version: 5.0.6
		Author: Dynamic.ooo

	Elementor
		Version: 3.14.1
		Author: Elementor.com

	Elementor Pro
		Version: 3.14.1
		Author: Elementor.com

	FacetWP
		Version: 4.2.2
		Author: FacetWP, LLC

	FacetWP - Elementor
		Version: 1.7
		Author: FacetWP, LLC

	FacetWP - Flyout menu
		Version: 0.8.1
		Author: FacetWP, LLC

	FacetWP - Map Facet
		Version: 1.0.3
		Author: FacetWP, LLC

	FacetWP - Multilingual support
		Version: 1.0.1
		Author: FacetWP, LLC

	JetElements For Elementor
		Version: 2.6.11
		Author: Crocoblock

	JetEngine
		Version: 3.2.2
		Author: Crocoblock

	JetEngine - dynamic tables builder
		Version: 1.0.7
		Author: Crocoblock

	JetFormBuilder
		Version: 3.0.9
		Author: Crocoblock

	JetFormBuilder Address Autocomplete
		Version: 1.0.7
		Author: Crocoblock

	JetMenu
		Version: 2.4.0
		Author: Crocoblock

	JetSearch
		Version: 3.1.0
		Author: Crocoblock

	JetSmartFilters
		Version: 3.1.2
		Author: Crocoblock

	JetTabs For Elementor
		Version: 2.1.22
		Author: Crocoblock

	Permalink Manager Pro
		Version: 2.4.1.2
		Author: Maciej Bis

	Post SMTP
		Version: 2.5.9
		Author: Post SMTP

	Post Snippets (free)
		Version: 4.0.4
		Author: Postsnippets

	Statify
		Version: 1.8.4
		Author: pluginkollektiv

	UpdraftPlus - Backup/Restore
		Version: 2.23.7.26
		Author: UpdraftPlus.Com, DavidAnderson

	Wordfence Security
		Version: 7.10.2
		Author: Wordfence

	WP All Import Pro
		Version: 4.7.6
		Author: Soflyy

	WP File Manager
		Version: 7.1.9
		Author: mndpsingh287

	WPML All Import
		Version: 2.3.0
		Author: OnTheGoSystems

	WPML Multilingual CMS
		Version: 4.6.3
		Author: OnTheGoSystems

	WPML SEO
		Version: 2.1.0
		Author: OnTheGoSystems

	WPML String Translation
		Version: 3.2.5
		Author: OnTheGoSystems

	Yoast SEO
		Version: 20.12
		Author: Team Yoast

== Fonctionnalités ==
	Custom Fonts: 0
	Custom Icons: 0

== Intégrations ==
	
	google_maps: Actif
	recaptcha_v3: Actif

== Expériences d’Elementor ==
	Sortie DOM optimisée: Inactif
	Amélioration du chargement des ressources: Actif par défaut
	Optimisation du chargement du CSS: Actif par défaut
	Icônes de police « inline »: Inactif par défaut
	Améliorations de l’accessibilité: Actif par défaut
	Points d’arrêt personnalisés supplémentaires: Actif par défaut
	admin_menu_rearrangement: Inactif par défaut
	Conteneur Flexbox: Actif
	Mettre à niveau la bibliothèque Swiper: Inactif par défaut
	Conteneur grille: Inactif par défaut
	Par défaut vers le Nouveau constructeur de thème: Actif par défaut
	En-tête et pied de page du thème Hello: Actif par défaut
	Barre supérieure de l’éditeur: Inactif par défaut
	Page d’atterrissage: Actif par défaut
	Éléments imbriqués: Inactif par défaut
	Accordéon imbriqué: Inactif par défaut
	Chargement différé des images d’arrière-plan: Inactif par défaut
	Guide de style global: Inactif par défaut
	Transitions des pages: Actif par défaut
	Notes: Actif par défaut
	Boucle: Actif par défaut
	Envois de formulaire: Actif par défaut
	Accroche de défilement: Actif par défaut
	Menu: Inactif par défaut

== Journal ==
	
JS: showing 20 of 41JS: 2023-07-26 22:31:15 [error X 3][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c120widgetsticky' 
JS: 2023-07-26 22:31:15 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c124widgetsticky' 
JS: 2023-07-26 22:31:15 [error X 3][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c136widgetsticky' 
JS: 2023-07-26 22:31:15 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c140widgetsticky' 
JS: 2023-07-26 22:31:15 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c150widgetsticky' 
JS: 2023-07-26 22:31:15 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c158widgetsticky' 
JS: 2023-07-26 22:31:15 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c176widgetsticky' 
JS: 2023-07-26 22:32:47 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c9026containersticky' 
JS: 2023-07-26 22:32:47 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c9025containersticky' 
JS: 2023-07-26 22:32:47 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c9027widgetsticky' 
JS: 2023-07-26 22:32:47 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c9031widgetsticky' 
JS: 2023-07-26 22:32:47 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c9037widgetsticky' 
JS: 2023-07-26 22:32:47 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c9041widgetsticky' 
JS: 2023-07-26 22:32:49 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c11035containersticky' 
JS: 2023-07-26 22:32:49 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c11034containersticky' 
JS: 2023-07-26 22:32:49 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c11046widgetsticky' 
JS: 2023-07-26 22:32:49 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c11050widgetsticky' 
JS: 2023-07-26 22:32:51 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c11036widgetsticky' 
JS: 2023-07-26 22:32:59 [error X 1][https://v3.le-rio.fr/wp-includes/js/jquery/jquery.min.js?ver=3.6.4:2:40999] Cannot create property 'guid' on string 'c11040widgetsticky' 
JS: 2023-07-26 23:52:47 [error X 6][https://v3.le-rio.fr/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.14.1:3:807522] Cannot read properties of undefined (reading 'localized') 

PHP: showing 1 of 1PHP: 2023-07-27 00:45:59 [notice X 3][../wp-content/plugins/elementor-pro/modules/posts/skins/skin-content-base.php::227] Trying to get property 'ID' of non-object [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

== Elementor - Compatibility Tag ==
	
	Dynamic Visibility for Elementor: Compatibilité non spécifiée
	Elementor Pro: Compatibilité non spécifiée
	FacetWP - Elementor: Compatibilité non spécifiée
	JetElements For Elementor: Compatibilité non spécifiée
	JetTabs For Elementor: Compatibilité non spécifiée

== Elementor Pro - Compatibility Tag ==

**Comment by nicholaszein** [comment]
Hi @ThomasCommpagnie! 

Thank you for your clarification! 

You are correct, I forgot to set the `Skin` to `Slideshow`. Apologies for the mistake. 🙏 

After testing it with this skin I can confirm the thumbnails do not show the dynamic images:


https://github.com/elementor/elementor/assets/113993460/b6570100-a369-4b9f-83fd-62d6c52ac907



🏷️ I've labeled this issue and our team will take a look at it. If there is indeed a problem, they'll work on fixing it. We can't say exactly when it'll be sorted out, but we kindly ask for your patience. Thanks for understanding!

Best regards

**Comment by ThomasCommpagnie** [comment]
Hi @nicholaszein, no problem, I wasn't very clear at the beggining too.

Thank you for your help ! Hope the tech team can fix it :) 

Best regards,

**Comment by Vallaf** [comment]
I have the same problem, has it been corrected ?

**Comment by ThomasCommpagnie** [comment]
> I have the same problem, has it been corrected ?

For the moment, the issue isn't fixed yet. Hope the team can have a look soon on this because it is very annoying in order to build dynamic websites...

**Comment by agentjblack** [comment]
We are experiencing this for a client site. We have ACF Gallery fields in most of the website's pages and the media gallery will not display thumbnails. Eager to see an update on this issue. Thank you guys!

**Comment by Lucania001** [comment]
Also waiting on a fix for this one. ACF fields, dynamic on media carrousel will not show thumbnails

**Comment by schmetterwurm** [comment]
Same problem for me. If Anyone have a fix pleas let me know <

**Comment by JustinBizPro** [comment]
Today is 10, Feb 2024.
this bug is still there, if anyone knows how to fix it please let me know, thanks.

![ACF Media Thumbnail Bug](https://github.com/elementor/elementor/assets/159564487/53d46a5d-5d78-445e-8a3c-001788674a1a)

**Comment by cedricDevWP** [comment]
Idem :)

**Comment by Hirello** [comment]
Hello I'm French and I had the same problem. I've made this code, it might help you.
It solve the problem of the "placeholder" thumbnails and also empty ACF Field.

It could be better but it work.

Put the entire code in your functions.php :
```

function str_replace_once($search, $replace, $subject) {
$pos = strpos($subject, $search);
if ($pos !== false) {
return substr_replace($subject, $replace, $pos, strlen($search));
}
return $subject;
}
function add_message_before_elementor_slideshow($widget_content, $widget) {
if ($widget->get_name() == 'media-carousel') {
	`$widget_content2 = $widget_content;`

	`$patternEmptySlide = 'style="background-image: url(&#039;&#039;)"';`
$replacementWithNothing = 'style="display:none;"';
	`$widget_content2 = str_replace($patternEmptySlide, $replacementWithNothing, $widget_content2, $count);`

	`$slides = $widget->get_settings('slides');`

	`foreach ($slides as $index => $slide) {`
$image_url = '';
		`$image_thumbnail = '';`
if (isset($slide['__dynamic__']['image'])) {
//Photo ACF for this slide
preg_match('/settings="([^"]+)"/', $slide['__dynamic__']['image'], $matches);
if (!empty($matches[1])) {
// Get acf field key refer to this slide
$settings = json_decode(urldecode($matches[1]), true);
$acf_field_key = $settings['key'];
if (strpos($acf_field_key, ':') !== false) {
list($field_key, $field_name) = explode(':', $acf_field_key);
$acf_image_infos = get_field($field_name);
if ($acf_image_infos) {
//ABLE TO GET IMAGE INFOS
$image_url = esc_url($acf_image_infos['url']);
$image_thumbnail = esc_url($acf_image_infos['sizes']['medium']);
} else {
//UNABLE TO GET IMAGE INFOS
$image_url = 'to_remove';
$image_thumbnail = 'to_remove';
}
}
}
} elseif (isset($slide['image']['url'])) {
//Photo not ACF for this slide
$image_url = esc_url($slide['image']['url']);
$thumbnail_info = wp_get_attachment_image_src($slide['image']['id'], 'thumbnail');
$image_thumbnail = esc_url($thumbnail_info[0]);
}
		`// Replace the thumbnail URL in the slide settings`

		`$pattern = site_url() . '/wp-content/plugins/elementor/assets/images/placeholder.png';`
$replacement = $image_thumbnail;
$widget_content2 = str_replace_once($pattern, $replacement, $widget_content2);
	`}`

	`$patternEmptySlide = 'style="background-image: url(&#039;to_remove&#039;)"';`
$replacementWithNothing = 'style="display:none;"';
	`$widget_content2 = str_replace($patternEmptySlide, $replacementWithNothing, $widget_content2, $count);`

	

	`$script = '<script type="text/javascript">`
document.addEventListener("DOMContentLoaded", function () {
// Sélectionner tous les éléments correspondants
var elementsToRemove = document.querySelectorAll(".swiper-slide .elementor-carousel-image[style=\'display:none;\']");
elementsToRemove.forEach(function(element) {
// Supprimer le parent .swiper-slide
var parentSlide = element.closest(".swiper-slide");
if (parentSlide) {
parentSlide.remove();
}
});
});
</script>';
$widget_content2 .= $script;
	`return $widget_content2;`
} else {
	`return $widget_content;`

`}`
}
add_filter('elementor/widget/render_content', 'add_message_before_elementor_slideshow', 10, 2);
```

**Comment by growthdyn-ross** [comment]
Hi all, I've also run into this bug (7/12/24). I came up with a javascript workaround that can be pasted into the elementor edit page:

<script defer="defer">
    let mainImgs = [...document.querySelectorAll('.elementor-skin-slideshow .elementor-swiper:nth-child(1) .swiper-container .swiper-wrapper .swiper-slide .elementor-carousel-image')];
    
    let previewImgs = [...document.querySelectorAll('.elementor-skin-slideshow .elementor-swiper:nth-child(2) .swiper-container .swiper-wrapper .swiper-slide .elementor-carousel-image')];
    
    mainImgs.forEach( (img, i) => {
        let bgImg = img.style.backgroundImage;
        
        previewImgs[i].style.backgroundImage = bgImg;
    })
</script>
